source/ada/lsp-ada_handlers.adb

```diff
@@ -1300,7 +1300,7 @@ package body LSP.Ada_Handlers is
             end if;
          else  --  If we are on a defining_name already
             Other_Part := Laltools.Common.Find_Next_Part
-              (Definition, Self.Trace);
+              (Definition, Self.Trace, Imprecise_Fallback => True);
```

Comment:
This change breaks the fact that the user gets warned when the results are not precise.

I suggest doing the following:

  - call `Find_Next_Part` with `Imprecise_Fallback => False`
  - if `Other_Part = No_Defining_Name`, set the boolean `Imprecise` to True and re-try `Find_Next_Part` with `Imprecise_Fallback => True`

Reply:
I will be slower: it is not recommended to call Find_Next_Part 2 times. Furthermore I would like to remove this Imprecise warning altogether.